### ISSUE TITLE
Workflow Controller panic on addParamToGlobalScope 

### ISSUE THREAD

**Issue by Farkal** [argoproj/argo-workflows#5843]
## Summary 

**What happened/what you expected to happen?**
Start workflow with some globalOutput
Workflow fail
The workflow controller go CrashLoopBackOff with panic

## Diagnostics

**What Kubernetes provider are you using?** 
Managed v1.20.6

**What version of Argo Workflows are you running?** 
v3.0.2

Code from the example: https://github.com/argoproj/argo-workflows/blob/master/examples/global-outputs.yaml with exit 1 to force crash of the workflow.
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: global-outputs-
spec:
  entrypoint: generate-globals
  onExit: consume-globals

  templates:
  - name: generate-globals
    steps:
    - - name: generate
        template: global-output
    - - name: consume-globals
        template: consume-globals

  # Template which produces a global parameter and artifact
  - name: global-output
    container:
      image: alpine:3.7
      command: [sh, -c]
      #args: ["sleep 1; echo -n hello world > /tmp/hello_world.txt"]
      args: ["sleep 1; exit 1"]
    outputs:
      parameters:
      # export a global parameter. The parameter will be programatically available in the completed
      # workflow object under: workflow.outputs.parameters
      - name: hello-param
        valueFrom:
          path: /tmp/hello_world.txt
        globalName: my-global-param
      # export a global artifact. The artifact will be programatically available in the completed
      # workflow object under: workflow.outputs.artifacts
      artifacts:
      - name: hello-art
        path: /tmp/hello_world.txt
        globalName: my-global-art

  # Once exported, global outputs are referenceable in later parts of the workflow.
  # In this example, the consume-globals template is invoked after the generate-globals step, as
  # well as in the onExit handler, and can reference the globals, my-global-param and my-global-art.
  - name: consume-globals
    steps:
    - - name: consume-global-param
        template: consume-global-param
      - name: consume-global-art
        template: consume-global-art
        arguments:
          artifacts:
          - name: art
            from: "{{workflow.outputs.artifacts.my-global-art}}"

  - name: consume-global-param
    inputs:
      parameters:
        - name: param
          value: "{{workflow.outputs.parameters.my-global-param}}"
    container:
      image: alpine:3.7
      command: [sh, -c]
      args: ["echo {{inputs.parameters.param}}"]

  - name: consume-global-art
    inputs:
      artifacts:
      - name: art
        path: /art
    container:
      image: alpine:3.7
      command: [sh, -c]
      args: ["cat /art"]
```

```
time="2021-05-06T08:43:55.533Z" level=info msg="Processing workflow" namespace=nrt workflow=global-outputs-tkknq
time="2021-05-06T08:43:55.535Z" level=info msg="Pod failed: Error (exit code 1)" displayName=generate namespace=nrt pod=global-outputs-tkknq-1660765316 templateName=global-output workflow=global-outputs-tkknq
time="2021-05-06T08:43:55.535Z" level=info msg="Setting node global-outputs-tkknq-1660765316 outputs" namespace=nrt workflow=global-outputs-tkknq
time="2021-05-06T08:43:55.535Z" level=info msg="Updating node global-outputs-tkknq-1660765316 status Pending -> Failed" namespace=nrt workflow=global-outputs-tkknq
time="2021-05-06T08:43:55.535Z" level=info msg="Updating node global-outputs-tkknq-1660765316 message: Error (exit code 1)" namespace=nrt workflow=global-outputs-tkknq
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19b12da]

goroutine 347 [running]:
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).addParamToGlobalScope(0xc0005f6fd0, 0xc00138fae0, 0xb, 0x0, 0x0, 0xc000950cc0, 0xc00138faf0, 0xf, 0x0, 0x0, ...)
    /go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:2560 +0x1ba
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).addOutputsToGlobalScope(0xc0005f6fd0, 0xc000a86700)
    /go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:2448 +0x118
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation.func1(0xc000079800)
    /go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:894 +0xe51
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation.func2(0xc00138fa90, 0xc00093bb30, 0xc0005f6fd0, 0x206bf80, 0xc000058020, 0xc00066a6c0, 0xc001478cc0, 0xc000950c00, 0xc000079800)
    /go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:948 +0x62
created by github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation
    /go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:946 +0x305
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by PacoDu** [comment]
We tried to downgrade argo to v2.12.11, same error.

v2.9.5 behaves as expected (no SIGSEGV)

**Comment by sarabala1979** [comment]
will take a look

**Comment by sarabala1979** [comment]
@PacoDu I am not able to reproduce with above example and v3.0.2
```
ow=global-outputs-pg9tr
time="2021-05-06T17:13:18.590Z" level=info msg="Updating node global-outputs-pg9tr-1748354537 status Running -> Failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.590Z" level=info msg="Updating node global-outputs-pg9tr-1748354537 message: Error (exit code 1)" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.592Z" level=info msg="Step group node global-outputs-pg9tr-3789702780 deemed failed: child 'global-outputs-pg9tr-1748354537' failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.592Z" level=info msg="node global-outputs-pg9tr-3789702780 phase Running -> Failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.592Z" level=info msg="node global-outputs-pg9tr-3789702780 message: child 'global-outputs-pg9tr-1748354537' failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.592Z" level=info msg="node global-outputs-pg9tr-3789702780 finished: 2021-05-06 17:13:18.5929416 +0000 UTC" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.593Z" level=info msg="step group global-outputs-pg9tr-3789702780 was unsuccessful: child 'global-outputs-pg9tr-1748354537' failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.595Z" level=info msg="Outbound nodes of global-outputs-pg9tr-1748354537 is [global-outputs-pg9tr-1748354537]" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.595Z" level=info msg="Outbound nodes of global-outputs-pg9tr is [global-outputs-pg9tr-1748354537]" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.595Z" level=info msg="node global-outputs-pg9tr phase Running -> Failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.596Z" level=info msg="node global-outputs-pg9tr message: child 'global-outputs-pg9tr-1748354537' failed" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.596Z" level=info msg="node global-outputs-pg9tr finished: 2021-05-06 17:13:18.5966067 +0000 UTC" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.596Z" level=info msg="Checking daemoned children of global-outputs-pg9tr" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.597Z" level=info msg="Running OnExit handler: consume-globals" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.597Z" level=info msg="Steps node global-outputs-pg9tr-2567452565 initialized Running" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.597Z" level=info msg="StepGroup node global-outputs-pg9tr-16303017 initialized Running" namespace=argo workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.597Z" level=error msg="Mark error node" error="unable to resolve references: Unable to resolve: {{workflow.outputs.artifacts.my-global-art}}" namespace=argo nodeName="global-outputs-pg9tr.onExit[0]" workflow=global-outputs-pg9tr
time="2021-05-06T17:13:18.597Z" level=info msg="node global-outputs-pg9tr-16303017 phase Running -> Error" namespace=argo workflow=global-outputs-pg9tr
```

**Comment by PacoDu** [comment]
@sarabala1979 We have patched the kustomize cluster deployment with `--secure=false` and the following configmap:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: workflow-controller-configmap
data:
  config: |
    containerRuntimeExecutor: docker
    metricsConfig:
      enabled: true
      path: /metrics
      persistence: {}
      port: 8080
    artifactRepository:
      archiveLogs: true
      s3:
        endpoint: <redacted>
        bucket: argo
        region: <redacted>
        insecure: false
        keyFormat: "artifacts\
          /{{workflow.creationTimestamp.Y}}\
          /{{workflow.creationTimestamp.m}}\
          /{{workflow.creationTimestamp.d}}\
          /{{workflow.name}}\
          /{{pod.name}}"
        accessKeySecret:
          name: <redacted>
          key: <redacted>
        secretKeySecret:
          name: <redacted>
          key: <redacted>
    persistence:
      archive: true
      clusterName: demo
      connectionPool:
        maxIdleConns: 100
        maxOpenConns: 0
      nodeStatusOffLoad: true
      postgresql:
        database: argo
        host: <redacted>
        passwordSecret:
          key: <redacted>
          name: <redacted>
        port: 5432
        tableName: argo_workflows
        userNameSecret:
          key: <redacted>
          name: <redacted>
    workflowDefaults:
      metadata:
        annotations:
          argo: workflows
      spec:
        serviceAccountName: argo
        ttlStrategy:
          secondsAfterSuccess: 30
          secondsAfterCompletion: 600
          secondsAfterFailure: 600
```

**Comment by sarabala1979** [comment]
@PacoDu  Can you try with `containerRuntimeExecutor: pns`?

**Comment by sarabala1979** [comment]
tried with `docker`
```
kflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="Updating node global-outputs-lntfc-1277992738 status Pending -> Failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="Updating node global-outputs-lntfc-1277992738 message: Error (exit code 1)" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="Step group node global-outputs-lntfc-2926438681 deemed failed: child 'global-outputs-lntfc-1277992738' failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="node global-outputs-lntfc-2926438681 phase Running -> Failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="node global-outputs-lntfc-2926438681 message: child 'global-outputs-lntfc-1277992738' failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="node global-outputs-lntfc-2926438681 finished: 2021-05-06 17:52:30.113878153 +0000 UTC" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="step group global-outputs-lntfc-2926438681 was unsuccessful: child 'global-outputs-lntfc-1277992738' failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="Outbound nodes of global-outputs-lntfc-1277992738 is [global-outputs-lntfc-1277992738]" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="Outbound nodes of global-outputs-lntfc is [global-outputs-lntfc-1277992738]" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="node global-outputs-lntfc phase Running -> Failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.113Z" level=info msg="node global-outputs-lntfc message: child 'global-outputs-lntfc-1277992738' failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc finished: 2021-05-06 17:52:30.114001046 +0000 UTC" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Checking daemoned children of global-outputs-lntfc" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Running OnExit handler: consume-globals" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Steps node global-outputs-lntfc-798760716 initialized Running" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="StepGroup node global-outputs-lntfc-1499088614 initialized Running" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=error msg="Mark error node" error="unable to resolve references: Unable to resolve: {{workflow.outputs.artifacts.my-global-art}}" namespace=default nodeName="global-outputs-lntfc.onExit[0]" workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-1499088614 phase Running -> Error" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-1499088614 message: unable to resolve references: Unable to resolve: {{workflow.outputs.artifacts.my-global-art}}" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-1499088614 finished: 2021-05-06 17:52:30.114590462 +0000 UTC" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="step group global-outputs-lntfc-1499088614 was unsuccessful: unable to resolve references: Unable to resolve: {{workflow.outputs.artifacts.my-global-art}}" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Outbound nodes of global-outputs-lntfc-798760716 is []" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-798760716 phase Running -> Failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-798760716 message: unable to resolve references: Unable to resolve: {{workflow.outputs.artifacts.my-global-art}}" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="node global-outputs-lntfc-798760716 finished: 2021-05-06 17:52:30.114675306 +0000 UTC" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Checking daemoned children of global-outputs-lntfc-798760716" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Updated phase Running -> Failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Updated message  -> child 'global-outputs-lntfc-1277992738' failed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Marking workflow completed" namespace=default workflow=global-outputs-lntfc
time="2021-05-06T17:52:30.114Z" level=info msg="Checking daemoned children of " namespace=default workflow=global-outputs-lntfc
```

**Comment by PacoDu** [comment]
I confirm, the issue is fixed with 3.0.7. Thanks !
